airbyte-workers/src/main/java/io/airbyte/workers/temporal/sync/PersistStateActivityImpl.java

```diff
@@ -39,4 +48,17 @@ public boolean persist(final UUID connectionId, final StandardSyncOutput syncOut
     }
   }
 
+  private static void validateStreamStates(final StateWrapper state, final ConfiguredAirbyteCatalog configuredCatalog) {
+    final List<StreamDescriptor> stateStreamDescriptors =
+        state.getStateMessages().stream().map(stateMessage -> stateMessage.getStream().getStreamDescriptor()).toList();
```

Comment:
Here you are grabbing the `stateMessages` from the state wrapper, which means we are assuming that this is a `STREAM` state. But this may not be the case, because `isMigration` just checks that the new state is anything besides `LEGACY`. So we'll need to either change this to only call `validateStreamStates` if the new state type is `STREAM`, or this method will need to be updated to work for any non-LEGACY state type

Reply:
This is only a concern when the new state type is stream, right? I think we said that but just want to make sure.

Reply:
Yep I believe so